capactio/capact
# Validate Type for storage backends

## Description

Changes proposed in this pull request:
- adds Type validation for storage backends 

## Testing
There was created a new generic storage backend schema in this [PR](https://github.com/capactio/hub-manifests/pull/67).

1. Checkout this PR
2. Create cluster using `DISABLE_MONITORING_INSTALLATION=true make dev-cluster `
3. Change manifests source to my branch:
```
kubectl set env deploy/capact-hub-public -n capact-system -c hub-public-populator MANIFESTS_SOURCES="github.com/mkuziemko/hub-manifests?ref=validate_storage_schema"
```
4. Build capact CLI `make build-tool-cli`
5. Execute tests on the hub-manifests
`capact manifests validate manifests/ -r --server-side`
6. Try to edit `cap.type.capactio.capact.validation.schema` by removing  `url` from `required` and repeat step 5. It should return validation errors.


There was also added a UT for this functionality. 

## Related issue(s)
- https://github.com/capactio/capact/issues/691
